sonar-db/src/main/java/org/sonar/db/version/CreateIndexBuilder.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.db.version;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.sonar.db.dialect.Dialect;
+import org.sonar.db.dialect.MySql;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Collections.singletonList;
+import static org.sonar.db.version.Validations.validateIndexName;
+import static org.sonar.db.version.Validations.validateTableName;
+
+public class CreateIndexBuilder {
+
+  private static final int MAX_LENGTH_ON_MYSQL = 255;
+
+  private final Dialect dialect;
+  private final List<ColumnDef> columns = new ArrayList<>();
+  private String tableName;
+  private String indexName;
+  private boolean unique = false;
+
+  public CreateIndexBuilder(Dialect dialect) {
+    this.dialect = dialect;
+  }
+
+  /**
+   * Required name of table on which index is created
+   */
+  public CreateIndexBuilder setTable(String s) {
+    this.tableName = s;
+    return this;
+  }
+
+  /**
+   * Required name of index. Name must be unique among all the tables
+   * of the schema.
+   */
+  public CreateIndexBuilder setName(String s) {
+    this.indexName = s;
+    return this;
+  }
+
+  /**
+   * By default index is NOT UNIQUE (value {@code false}).
+   */
+  public CreateIndexBuilder setUnique(boolean b) {
+    this.unique = b;
+    return this;
+  }
+
+  /**
+   * Add a column to the scope of index. Order of calls to this
+   * method is important and is kept as-is when creating the index.
+   * The attributes used from {@link ColumnDef} are the name, the type
+   * and the length (in case of VARCHAR). Other attributes are ignored.
+   */
+  public CreateIndexBuilder addColumn(ColumnDef column) {
+    columns.add(column);
```

Comment:
to ensure no `null` can be added to the list (check in `build` method only ensures `columns` is not empty), which is a case of invalid data, you should had a check on the argument of this method